op-e2e/interop/supersystem.go

```diff
@@ -0,0 +1,622 @@
+	L2Geth(network string) *geth.GethInstance
+	// get the L2 geth client for a network
+	L2GethClient(network string) *ethclient.Client
+	// get the secret for a network and role
+	L2OperatorKey(network string, role devkeys.ChainOperatorRole) ecdsa.PrivateKey
+	// get the list of network IDs
+	L2IDs() []string
+	// register a username to an account on all L2s
+	AddUser(username string)
+	// get the user key for a user on an L2
+	UserKey(id, username string) ecdsa.PrivateKey
+	// send a transaction on an L2 on the given network, from the given user
+	SendL2Tx(network string, username string, applyTxOpts op_e2e.TxOptsFn) *types.Receipt
+	// get the address for a user on an L2
+	Address(network string, username string) common.Address
+}
+
+// NewSuperSystem creates a new SuperSystem from a recipe. It creates an interopE2ESystem.
+func NewSuperSystem(t *testing.T, recipe *interopgen.InteropDevRecipe) SuperSystem {
+	s2 := &interopE2ESystem{recipe: recipe}
+	s2.prepare(t)
+	return s2
+}
+
+// interopE2ESystem implements the SuperSystem interface
+// it prepares network resources and provides access to them
+// the functionality is broken down into smaller functions so that
+// the system can be prepared iteratively if desired
+type interopE2ESystem struct {
+	t               *testing.T
+	recipe          *interopgen.InteropDevRecipe
+	logger          log.Logger
+	hdWallet        *devkeys.MnemonicDevKeys
+	worldDeployment *interopgen.WorldDeployment
+	worldOutput     *interopgen.WorldOutput
+	beacon          *fakebeacon.FakeBeacon
+	l1              *geth.GethInstance
+	l2s             map[string]l2Set
+	l2GethClients   map[string]*ethclient.Client
+	supervisor      *supervisor.SupervisorService
+	superClient     *sources.SupervisorClient
+}
+
+// l2Set is a set of resources for an L2 chain
+type l2Set struct {
+	chainID      *big.Int
+	opNode       *opnode.Opnode
+	l2Geth       *geth.GethInstance
+	proposer     *l2os.ProposerService
+	batcher      *bss.BatcherService
+	operatorKeys map[devkeys.ChainOperatorRole]ecdsa.PrivateKey
+	userKeys     map[string]ecdsa.PrivateKey
+}
+
+// prepareHDWallet creates a new HD wallet to derive keys from
+func (s *interopE2ESystem) prepareHDWallet() *devkeys.MnemonicDevKeys {
+	hdWallet, err := devkeys.NewMnemonicDevKeys(devkeys.TestMnemonic)
+	require.NoError(s.t, err)
+	return hdWallet
+}
+
+// prepareWorld creates the world configuration from the recipe and deploys it
+func (s *interopE2ESystem) prepareWorld() (*interopgen.WorldDeployment, *interopgen.WorldOutput) {
+	// Build the world configuration from the recipe and the HD wallet
+	worldCfg, err := s.recipe.Build(s.hdWallet)
+	require.NoError(s.t, err)
+
+	// create a logger for the world configuration
+	logger := s.logger.New("role", "world")
+	require.NoError(s.t, worldCfg.Check(logger))
+
+	// create the foundry artifacts and source map
+	foundryArtifacts := foundry.OpenArtifactsDir("../../packages/contracts-bedrock/forge-artifacts")
```

Comment:
I think its generally better to allow these to be configurable, perhaps setting this value within an init function. @mslipper built some tooling to handle the path to artifacts where it can accept an arbitrary URI, if we could set an env var then we could also run this against a release of the contracts that is hosted in the cloud. This may not work well if we depend on the source maps since the releases only include the artifacts, and it looks like the source map tooling depends on the source code directly rather than the artifacts

Reply:
Sure, I can pull this out to the test level at least, and we can look at different ways to load it in the future.